docs/source/dsa.md

```diff
@@ -0,0 +1,80 @@
+# Digital Slide Archive
+
+The [Digital Slide Archive][1] (DSA) is "a containerized web-based platform for the analysis, visualization, management and
+annotation of whole-slide digital pathology imaging data".
+It is an open-source project based on [Kitware][11]'s data management platform [Girder][10].
+
+## Azure deployment
+
+We have deployed the DSA on [Azure][2] to visualize our data and interpret our models and experiments.
+Below are instructions to replicate our deployment using your own data.
+
+### Host
+
+The first step is [creating a Linux virtual machine][4] (VM) to host the DSA.
+The code can be [downloaded from GitHub][5] and run using [Docker][6].
+By default, the application runs at port 8080.
+To use HTTPS, an SSL certificate must be obtained and port 443 may be specified instead within the Docker configuration.
+The DSA uses [Cherrypy][8] as the underlying server engine.
+Therefore, Cherrypy must be [configured][9] to use the SSL certificate installed in the VM.
+Ports in a VM can be opened using [network security group rules][7].
+
+### Storage
+
+The datasets we use are securely stored in [Azure Blob Storage][3] containers.
+Containers can be mounted on the host machine using [BlobFuse][13] and connected to Girder using [assetstores][12].
+
+### Authentication
+
+We have [added][14] Microsoft as a provider in the [OAuth2 Login][15] plugin for Girder.
+An app must be [registered][16] so that [Azure Active Directory][17] (Azure AD) can be used for authentication.
+
+### Authorization
+
+User [permissions][18] can be set to different data collections in an assetstore, to maximize protection of sensitive data.
+
+## Visualizing Azure Machine Learning results
+
+The [Girder RESTful API][19] may be used to upload annotations to DSA items programmatically.
+An example of a use case is creating annotations to visualize attention maps generated by a deep learning model.
+The `girder` module in this repository includes tools to download training results from Azure Machine Learning (Azure ML) and
+upload annotations to a deployed DSA.
+For example:
+
+```shell
+SCRIPT="hi-ml-histopathology/src/histopathology/utils/girder.py"
+python $SCRIPT --run-id "Experiment_transformer_Gleason_run_0" --rescale
```

Comment:
Out of scope for this PR, but just to say that we have not used "-" as the separator for commandline args throughout most of the codebase, but "_". This may be frowned upon as not pythonic, but has the huge advantage that we can use auto-generated parsers, which eliminate a classic source of errors.